.github/workflows/run-tests.yml

```diff
@@ -27,5 +26,49 @@ jobs:
           path: ~/.m2
           key: ${{ runner.os }}-m2-${{ hashFiles('**/pom.xml') }}
           restore-keys: ${{ runner.os }}-m2
-      - name: Run integration tests
-        run: mvn clean verify
+      - name: Run tests
+        run: mvn clean package
+      - name: Upload Artifact
+        uses: actions/upload-artifact@v2
+        with:
+          name: target
+          path: target
+          retention-days: 5
+  build:
+    needs: test
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@5a4ac9002d0be2fb38bd78e4b4dbde5606d7042f
+        with:
+          fetch-depth: '0'
+      - name: Download a single artifact
+        uses: actions/download-artifact@v2
+        with:
+          name: target
+          path: target
+      - name: Build docker image
+        run: |
+          ls -la
+          docker build -t govukpay/webhooks .
+  push:
+    needs: build
+    runs-on: ubuntu-latest
+    permissions:
+      id-token: write
+      contents: read
+    steps:
+      - run: sleep 5
+      - name: Fake push
+        run: |
+          export AWS_ROLE_ARN=arn:aws:iam::223851549868:role/github_actions_role
```

Comment:
It might be overkill but I wonder if we include the ARN as a GitHub secret, that way if it does get out there somehow we're not fussed but otherwise it's only available to project contributors.

All of the linked posts and references just include it directly and a brief search doesn't bring up a clear best practice for keeping them hidden (other than favouring templating them in from the thing provisioning them which doesn't apply in this case)

Reply:
Yeah probably a good plan. A bit security by obscurity, but given it's not really any  effort might as well.